src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -319,6 +325,141 @@ def fuzz(self,
                                 artifact_prefix, additional_args)
 
 
+class FuchsiaQemuLibFuzzerRunner(new_process.ProcessRunner, LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform)."""
+
+  def __init__(self, executable_path, default_args=None):
+    self.fuchsia_qemu_command_template = [
+        '{qemu}', '-D', '/tmp/qemustderr', '-m', '2048', '-nographic',
+        '-kernel', '{kernel}', '-initrd', '{initrd}', '-smp', '4', '-drive',
+        'file={drive},format=raw,if=none,id=blobstore', '-device',
+        'virtio-blk-pci,drive=blobstore', '-monitor', 'none', '-append',
+        'kernel.serial=legacy TERM=dumb', '-machine', 'q35', '-enable-kvm',
+        '-display', 'none', '-cpu', 'host,migratable=no', '-netdev',
+        'user,id=net0,net=192.168.3.0/24,dhcpstart=192.168.3.9,host=192.168.3.2,hostfwd=tcp::{portnum}-:22',
+        '-device', 'e1000,netdev=net0,mac=52:54:00:63:5e:7b', '-L',
+        '{sharefiles}'
+    ]
+
+    self.fuchsia_ssh_command_template = [
+        'ssh', '-i', '{identity_file}', '-o', 'StrictHostKeyChecking no',
+        'localhost', '-p', '{portnum}'
+    ]
+
+    self.fuchsia_bucket_name = 'fuchsia_on_clusterfuzz_resources_v1'
+
+    self.fuchsia_gsutil_command_template = [
+        'gsutil', 'cp', '-r', '{fuchsia_resources_path}',
+        '{local_resources_path}'
+    ]
+
+    # This initialization routime assumes that the GCS bucket contains the standard Fuchsia SDK, as well as:
+    # * /qemu-for-fuchsia/*
+    # * /.ssh/*
+    resources_path = os.path.join(os.getcwd(), self.fuchsia_bucket_name)
+
+    shell.create_directory_if_needed(resources_path)
+
+    if 'FUCHSIA_RESOURCES_PATH' not in os.environ:
+      logs.log_error("Could not find path for remote Fuchsia resources bucket (FUCHSIA_RESOURCES_PATH)!")
+
+    subbed_fuchsia_gsutil_command = [
+        param.replace("{fuchsia_resources_path}",
+                      environment.get_value('FUCHSIA_RESOURCES_PATH', ''))
+        .replace("{local_resources_path}", resources_path)
+        for param in self.fuchsia_gsutil_command_template
+    ]
+    subprocess.call(subbed_fuchsia_gsutil_command)
```

Comment:
`subprocess.check_call` would be better, to make sure we fail in case of an error and report an exception to StackDriver?

Reply:
Actually, I think the preferred way is to use `process = new_process.ProcessRunner(executable_path, default_args=None)` followed by `result = process.run_and_wait(additional_args=None)` and checking the `result.return_code` afterwards